dangerzone/gui/__init__.py

```diff
@@ -123,6 +123,7 @@ def open_files(filenames: List[str] = []) -> None:
     # Check for updates
     log.debug("Setting up Dangezone updater")
     updater = UpdaterThread(dangerzone)
+    updater.update_check_toggled.connect(window.refresh_updates_checkbox)
```

Comment:
Personally, I'd replace everything with these three lines:

```diff
diff --git a/dangerzone/gui/__init__.py b/dangerzone/gui/__init__.py
index f0249ac..c35d232 100644
--- a/dangerzone/gui/__init__.py
+++ b/dangerzone/gui/__init__.py
@@ -132,6 +132,10 @@ def gui_main(
     else:
         log.debug("Will not check for updates, based on updater settings")
 
+    # Ensure the status of the toggle updates checkbox is updated, after the user is
+    # prompted to enable updates.
+    window.toggle_updates_action.setChecked(bool(updater.check))
+
     if filenames:
         open_files(filenames)
 
```

I'd suggest against including the signal mechanism, since the main window has not started yet, meaning that we are not constrained by the limitations of the Qt event loop.